### DISCUSSION TITLE
Feedback on tutorial

### DISCUSSION THREAD

**Discussion by staropram** [moj-analytical-services/splink#1708] [General]
Hi, first of all let me say thanks for creating such an amazing package with comprehensive documentation and tutorials. You really have all done an excellent job on it. Go Team. I've gone through the tutorial 3 times and as such have found a couple of typos and have some suggestions.

1. For the getting started page (or perhaps as intro to tutorial), to encourage good library hygene and to ensure things work first time for new users, could suggest a workflow like this:

```
python3 -m venv ~/venvs/splink_tutorial
. ~/venvs/splink_tutorial/bin/activate
pip install splink notebook
jupyter notebook
```

Then they'd have a notebook setup that should just work. Perhaps that could be at the start of the tutorial and this might reduce support requests coming in with broken setups.

2. in the "Blocking" part in the section entitled "Counting the number of  comparisons created by a single blocking rule" it references the SQL of the blocking rules examined.

blocking_rule_1.blocking_rule_sql

this should be

blocking_rule_1.sql

etc

3. In the "Estimating model parameters" section 
some of the the code examples have retain escaped sequences for < such as \&lt; so the code examples won't copy and paste without modification

same issue in "Clustering" in the "Predicting results" section

4. In "Quality assurance" section, the "load in labels" bit should probably put df_labels.head(5) as the last line so it is shown in the notebook

That's it. Amazing stuff, cheers!

**Comment by RossKen** [comment]
Hey @staropram - thanks so much for the feedback! I will open up a PR now to make these changes 😊

**Reply by staropram** [reply]
Cheers!

**Reply by RobinL** [reply]
@staropram just to note

`blocking_rule_1.blocking_rule_sql` this should be `blocking_rule_1.sql`

That's my fault.  We have recently made a change so that in future it _will be_ `blocking_rule_1.blocking_rule_sql` but in the version of code you're using it's not yet.  (If you use Splink on master, or wait for the next release, that's what it will be)

Only noting to say sorry for the trouble, and if you see Ross hasn't fixed this, this is why (he's not ignoring the feedback)

**Reply by staropram** [reply]
Ah I see, makes sense, thanks for the clarification. Would you recommend using Splink on master?

**Reply by RobinL** [reply]
There's no problem in doing so, but I' not aware of any major bugs or features that have recently been released, so I don't think you'll see any big practical differences in terms of functionality

The change in `blocking_rule_sql` is a 'under the hood' thing to make it easier for Splink developers to understand the backend code

**Reply by staropram** [reply]
Thanks
